lib/src/Pos/Generator/Block/Logic.hs

```diff
@@ -117,10 +118,12 @@ genBlock eos = do
                     Just <$> usingPrimaryKey leaderSK (genMainBlock slot (Right . swap <$> transCert))
   where
     genMainBlock slot proxySkInfo =
+        withCompileInfo def $
         lift $ createMainBlockInternal @SscGodTossing slot proxySkInfo >>= \case
             Left err -> throwM (BGFailedToCreate err)
             Right mainBlock -> verifyAndApply $ Right mainBlock
     verifyAndApply block =
+        withCompileInfo def $
```

Comment:
Looks strange to have 3 `withCompileInfo` in `genBlock`. Maybe you need to provide type signatures explicitly.

Reply:
I think this difference is mostly cosmetical, but i can do it, sure.

Reply:
The problem is that if `withCompileInfo` is applied to the same code multiple times with different values, something bad will happen, AFAIK (and it won't be prevented by compiler).
So having less `withCompileInfo` makes reasoning easier.
Later someone may change `def` to something else in one place and forget about another one.

Reply:
In general -- yes. This particular place -- no. But anyway, I agree.